CodeGeneration/Sources/Utils/CodeGenerationFormat.swift

```diff
@@ -96,6 +96,24 @@ public class CodeGenerationFormat: BasicFormat {
     let children = node.children(viewMode: .all)
     // Short tuple element list literals are presented on one line, list each element on a different line.
     if children.count > maxElementsOnSameLine {
+      let inMethodCallThatStartsOnNewline =
+        if let functionCallExpr = node.parent?.as(FunctionCallExprSyntax.self),
+          let memberAccessExpr = functionCallExpr.calledExpression.as(MemberAccessExprSyntax.self),
+          startsOnNewline(memberAccessExpr.period)
+        {
+          true
+        } else {
+          false
+        }
```

Comment:
If you mad `startsOnNewline` an extension on `SyntaxProtocol` this could be done without an `if` expression:

```swift
node.parent?.as(FunctionCallExprSyntax.self).calledExpression.as(MemberAccessExprSyntax.self)?.startsOnNewline ?? false
```

---

Why does this only apply if the called expression is a `MemberAccessExprSyntax`? Shouldn’t the same be true for top level function calls as well?

Reply:
> If you mad `startsOnNewline` an extension on `SyntaxProtocol` this could be done without an `if` expression:
> 
> ```swift
> node.parent?.as(FunctionCallExprSyntax.self).calledExpression.as(MemberAccessExprSyntax.self)?.startsOnNewline ?? false
> ```

Done 👍

> Why does this only apply if the called expression is a `MemberAccessExprSyntax`? Shouldn’t the same be true for top level function calls as well?

If you mean top level functions calls by specifying the module, they also parse into `MemberAccessExprSyntax`. Example:
```swift
Swift
  .print(
    "1",
    "2",
    separator: ",",
    terminator: ";"
  )
``` 
<img width="765" alt="image" src="https://github.com/user-attachments/assets/a5580897-ceec-4438-801a-94e54ae10f2f">

Reply:
I mean if you have something like

```swift
print(
  "1", 
  "2"
)
```

Or do we format that correctly already?

Reply:
> Or do we format that correctly already?

Yes we do.

Reply:
Oh, great. Thanks for confirming